R/zchunk_L202.an_input.R

```diff
@@ -0,0 +1,646 @@
+      # not every region/technology/year has a match, so need to use left_join
+      left_join(L202.ag_Feed_Mt_R_C_Y.mlt, by = c("region", "technology" = "GCAM_commodity", "year")) %>%
+      mutate(calOutputValue = round(value, aglu.DIGITS_CALOUTPUT)) %>%
+      # the DDGS/feedcake rows at this point are all missing values, as they are not
+      # available in the historical years; set them to zero
+      replace_na(list(calOutputValue = 0)) %>%
+      # subsector and technology shareweights (subsector requires the year as well)
+      mutate(share.weight.year = year,
+             subs.share.weight = if_else(calOutputValue > 0, 1, 0),
+             tech.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["StubTechProd"]])) ->
+      L202.StubTechProd_in
+
+    # L202.Supplysector_an: generic animal production supplysector info (159-162)
+    A_an_supplysector %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["Supplysector"]]), GCAM_region_names) ->
+      L202.Supplysector_an
+
+    # L202.SubsectorAll_an: generic animal production subsector info (164-167)
+    A_an_subsector %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["SubsectorAll"]]), GCAM_region_names) ->
+      L202.SubsectorAll_an
+
+    # L202.StubTech_an: identification of stub technologies for animal production (169-171)
+    A_an_technology %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]], GCAM_region_names) %>%
+      rename(stub.technology = technology) ->
+      L202.StubTech_an
+
+    # L202.StubTechInterp_an: shareweight interpolation for animal production technologies (173-175)
+    A_an_technology %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["TechInterp"]], GCAM_region_names) %>%
+      rename(stub.technology = technology) ->
+      L202.StubTechInterp_an
+
+    # L202.StubTechProd_an: animal production by technology and region (177-199)
+    A_an_technology %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]], GCAM_region_names) %>%
+      mutate(stub.technology = technology) %>%
+      repeat_add_columns(tibble(year = BASE_YEARS)) %>%
+      left_join_error_no_match(L202.an_Prod_Mt_R_C_Sys_Fd_Y.mlt,
+                               by = c("region", "supplysector" = "GCAM_commodity",
+                                      "subsector" = "system", "technology" = "feed",
+                                      "year")) %>%
+      mutate(calOutputValue = round(value, aglu.DIGITS_CALOUTPUT)) %>%
+      # subsector and technology shareweights (subsector requires the year as well)
+      mutate(share.weight.year = year,
+             subs.share.weight = if_else(calOutputValue > 0, 1, 0),
+             tech.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["StubTechProd"]])) ->
+      L202.StubTechProd_an
+
+    # Some subsectors have multiple technologies, so shareweights should be derived from aggregation
+    L202.StubTechProd_an %>%
```

Comment:
why do share weights have to come from an aggregation for this output and not the previous?